pkg/nsx/services/subnet/subnet.go

```diff
@@ -0,0 +1,146 @@
+package subnet
+
+import (
+	"os"
+	"sync"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var (
+	log                       = logger.Log
+	MarkedForDelete           = true
+	EnforceRevisionCheckParam = false
+	ResourceTypeSubnet        = common.ResourceTypeSubnet
+	NewConverter              = common.NewConverter
+)
+
+type SubnetService struct {
+	common.Service
+	SubnetStore *SubnetStore
+}
+
+var subnetService *SubnetService
+
+// GetSubnetService get singleton SubnetService instance, subnet/subnetset controller share the same instance.
+// GetSubnetService isn't thread-safe.
+func GetSubnetService(service common.Service) *SubnetService {
```

Comment:
No lock?  refer [https://refactoring.guru/design-patterns/singleton/go/example](url)

Reply:
There is no lock as the function wasn't expected to be thread-safe. But it's better to add a lock for this, thanks.

Reply:
Lock added.